### ISSUE TITLE
Subtitles does not work

### ISSUE THREAD

**Issue by hejman** [Diblo/KODI-Popcorn-Time#46]
Hi

I get no subtitles although i set #1 Preferred subtitle language in Add-on preferences.

I was under the impression that the addon would download subtitles from yify subs out of the box?

Is this supposed to work as a Subtitle service? 

Thanks in advance, great work!

**Comment by chakhdin** [comment]
hi, @hejman , can you please provide your logs with debugging enabled, please?

**Comment by hejman** [comment]
Sure thing

https://dl.dropboxusercontent.com/u/13784723/kodi.log

**Comment by chakhdin** [comment]
Are you sure you selected the subtitle language, because the url in the log shows "none" (right after /play/):
plugin://plugin.video.kodipopcorntime/play/none/magnet....

**Comment by hejman** [comment]
Yeah im sure
![screen shot 2015-09-16 at 23 16 49](https://cloud.githubusercontent.com/assets/11835166/9918801/48c214ba-5cc9-11e5-89ac-d08c3ba4d42a.png)

**Comment by hejman** [comment]
If you want me to I can try the alpha version out.

But that is also causing me trouble! :-( tried installing the below zip but it causes me a structure error in Kodi.

https://dl.dropboxusercontent.com/u/13784723/plugin.video.kodipopcorntime.zip

Thanks for your assistance btw

**Comment by Diblo** [comment]
Hi @hejman 
I do not know why you have this problem on mac, and the problem with version 0.6.5 is that we are missing some debug information. However you can try KODI Popcorn Time version 0.7.0 Alpha, and see if the problem persists.
The reason you get the error "structure error" with version 0.7.0 Alpha is because KODI can not handle a MAC zipped file. To work around this problem please read issue [#28#issuecomment-134543966](https://github.com/Diblo/KODI-Popcorn-Time/issues/28#issuecomment-134543966) or download this [zip file](https://drive.google.com/file/d/0B3fNnjj64nUpZVRKNndnbFZqRFE/view?usp=sharing).

... we can take it in danish if it is easier?

@chakhdin
Do you have time to create a wiki page about creating a zip fil on mac, and one about how the debug version 0.6.5 and 0.7.0?

**Comment by Diblo** [comment]
https://github.com/Diblo/KODI-Popcorn-Time/wiki/How-To-and-Guides

**Comment by hejman** [comment]
@Diblo 
Hi thanks for your response, english is fine with me - and makes it easier for other users to find answers :-)

I finally manage to zip the ALPHA properly on mac - what a hassle.

Unfortunately 0.7.0 does not work at all on my machine, it gives an unknown error when attempting to launch a movie.

See log here:
https://dl.dropboxusercontent.com/u/13784723/kodi.log

Again thank you very much for your work on this!

**Comment by hejman** [comment]
I see no mention of the subtitle_yify provider in Kodi nor in the logs - is it loaded properly?

```
10:32:51 T:123145306673152   ERROR: [plugin.video.kodipopcorntime] An unknown error has occurred
10:32:51 T:123145306673152   ERROR: [plugin.video.kodipopcorntime] Traceback (most recent call last):
10:32:51 T:123145306673152   ERROR: [plugin.video.kodipopcorntime]   File "/Users/USER/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/addon.py", line 276, in play
10:32:51 T:123145306673152   ERROR: [plugin.video.kodipopcorntime]     Player().play(url, kwargs["item"], kwargs['subtitle'], subtitle_provider)
10:32:51 T:123145306673152   ERROR: [plugin.video.kodipopcorntime]   File "/Users/USER/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/resources/lib/kodipopcorntime/player.py", line 168, in play
10:32:51 T:123145306673152   ERROR: [plugin.video.kodipopcorntime]     free_space = self._calculate_free_space(downloadpath)
10:32:51 T:123145306673152   ERROR: [plugin.video.kodipopcorntime]   File "/Users/USER/Library/Application Support/Kodi/addons/plugin.video.kodipopcorntime/resources/lib/kodipopcorntime/player.py", line 102, in _calculate_free_space
10:32:51 T:123145306673152   ERROR: [plugin.video.kodipopcorntime]     st = os.statvfs(download_path)
10:32:51 T:123145306673152   ERROR: [plugin.video.kodipopcorntime] OSError: [Errno 2] No such file or directory: ''
```

**Comment by chakhdin** [comment]
It's a bug I introduced recently. As a workaround, please, specify a download directory, instead of using the default value. 
I'll correct it later today. 

Cheers

**Comment by hejman** [comment]
That did the trick - thanks, the movie now loaded.

I still have problems with subtitles on all other movies than the first one returned by the listing "Most popular" "Age of Avengers".

Also i experienced extremely low fps followed by a crash when switching from the running film above to Entourage.

New log file https://dl.dropboxusercontent.com/u/13784723/kodi.log

**Comment by chakhdin** [comment]
@hejman , can you please check the current version of the plugin?
The error should be gone, even if you don't specify a custom download path in the settings.
I'm affraid that the only way to reset that setting is actually remove the line from ..\Kodi\userdata\addon_data\plugin.video.kodipopcorntime\settings.xml

Please, let me know.

**Comment by hejman** [comment]
@chakhdin I no longer have access to the Alpha branch - so I am unable to download the latest version

**Comment by Diblo** [comment]
Se if this help: https://github.com/Diblo/KODI-Popcorn-Time/wiki/Test-releases

**Comment by Diblo** [comment]
I have just released Alpha version 2 - You can downloade it from her: https://github.com/Diblo/KODI-Popcorn-Time/wiki/Test-releases#test-alpha-version-070

See if this works - Remember to read the note.

**Comment by hejman** [comment]
Will test it later this week, thanks Henrik :-)
- Very nice work you've done on this!

I used to use Pulsar with a yify provider, it's very heavy and often
crashes - also it has memory issues so I had to restart my Raspberry Pi
almost everytime i had been watching a movie.

Btw, I know a lot of people are looking for a yifysubtitles.com service for
Kodi.
- Perhaps you want to share yours?
http://forum.kodi.tv/showthread.php?tid=214725
and
http://forum.kodi.tv/showthread.php?tid=214879&highlight=yifysubtitles



Best regards

Kasper Nørskov
